cpu/kinetis/include/periph_cpu.h

```diff
@@ -130,9 +130,33 @@ typedef uint16_t gpio_t;
 #define PERIPH_TIMER_PROVIDES_SET
 
 /**
- * @brief   number of usable power modes
+ * @name    Kinetis power mode configuration
+ * @{
+ */
+#define PM_NUM_MODES    (4U)
```

Comment:
Is `KINETIS_PM_WAIT` the default (`idle`) state?
If you must put `#define PM_NUM_MODES (3U)` here. (Yes, this is confusing… #13475)

Reply:
If I'm understanding correctly, the consensus is that we want the highest enterable state to be what ARM calls `Sleep` which is where everything stays on but the CPU calls `wfi()`. That would mean PM can't block that mode which means PM can't enter what ARM calls `Run` mode which is effectively a while(1){} state. Is that right?

I'm not saying it's terribly useful for PM to enter that mode. I'm just making sure I understand. I've been using it this way for years and it never occurred to me to do otherwise.

I've added comments for each power mode to try to remove any confusion from differing terminologies as I'm not certain what you mean by "idle state". Hopefully now you can tell me if it's definitely wrong the way it is.

Reply:
Hm I think there is a misunderstanding.
From what I read, the default IDLE mode should be `KINETIS_PM_WAIT`. You can remove `case PM_NUM_MODES`, that is not needed.

But `PM_NUM_MODES` only counts the non-IDLE modes (that is all other modes: `KINETIS_PM_STOP`, `KINETIS_PM_VLPS`, `KINETIS_PM_LLS` - those are 3).

Reply:
Ok I think that satisfies my uncertainty about how it should be. I've pushed an update to change it as you said.

> Why it often still works with the 'wrong' number is that many implementations have a `default:` case at their normal IDLE mode, so it will also catch all 'higher' modes.

In this case it worked because there was a `PM_NUM_MODES:` case to handle it, but that wasn't how it was supposed to work.